qt/python/mantidqt/widgets/instrumentview/instrument_view.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+import functools
+from enum import IntEnum
+from mantidqt.utils.qt.qappthreadcall import QAppThreadCall
+from mantidqt.widgets.instrumentview.presenter import InstrumentViewPresenter
+
+
+def safe_qthread(func):
```

Comment:
Would `force_method_calls_to_qapp_thread` be a better choice?

https://github.com/mantidproject/mantid/blob/f4f45f0b7993293c83ec8e5a9b159f0314ea3035/qt/python/mantidqt/utils/qt/qappthreadcall.py#L85

Reply:
It could.  We found it rather late after @KedoKudo implemented _pyInstrumentView_

Reply:
I was thinking about switching to this method, but it seems like it is only used here https://github.com/mantidproject/mantid/blob/039b87326ad3138f4fe5a92acfc625f69e4a36ae/qt/applications/workbench/workbench/plotting/figuremanager.py#L494
and the call is also wrapped in another `QAppThreadCall`, which I cannot figure out the reason behind it.
So I decided not make the switch in this PR.

Reply:
I would advise switching to this method that Steve mentions. It will save us having more things that are done slightly differently in different places.
The [call](https://github.com/mantidproject/mantid/blob/039b87326ad3138f4fe5a92acfc625f69e4a36ae/qt/applications/workbench/workbench/plotting/figuremanager.py#L504) must first ensure that the call to `_new_figure_manager_given_figure_impl` is performed on the QApp thread and then this internal function creates a new `FigureManagerWorkbench` and the `force_method_calls_to_qapp_thread` method then takes this object and forces each method to be called on the QApp thread.

Reply:
The new implementation uses the suggested method to create an instance of instrument view presenter, along with some common accessors exposed to the top level via duck-typing.
Due to some unknown strange issue, the two methods, `setMinValue` and `setMaxValue`, need to be wrapped in `QAppThreadCall` explicitly to avoid a segmentation fault on RHEL_7.

Reply:
`setMinValue` and `setMaxValue` are Qt/C++ methods of InstrumentWidgetRenderTab.  Can this be the cause?

Reply:
The actual implementation of these two calls are as follows, so maybe something about the `m_colorBarWidget` is causing the issue.
https://github.com/mantidproject/mantid/blob/760ef2ae9c914b2b2bdd0087b6ba73160624dc8a/qt/widgets/instrumentview/src/InstrumentWidgetRenderTab.cpp#L494-L500

Reply:
The other Qt/C++ methods of `InstrumentWidgetRenderTab` work out-of-the-box (setAxis, etc.), and this issue only occurs on RHEL_7.
This is really strange and confusing :confused: